x-pack/legacy/plugins/monitoring/public/components/setup_mode/tooltip.js

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { get } from 'lodash';
+import {
+  EuiBadge,
+  EuiFlexItem,
+  EuiToolTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { getIdentifier } from './formatting';
+
+export function SetupModeTooltip({ setupModeData, badgeClickAction, productName }) {
+  if (!setupModeData) {
+    return null;
+  }
+
+  const {
+    totalUniqueInstanceCount,
+    totalUniqueFullyMigratedCount,
+    totalUniquePartiallyMigratedCount
+  } = setupModeData;
+  const allMonitoredByMetricbeat = totalUniqueInstanceCount > 0 &&
+    (totalUniqueFullyMigratedCount === totalUniqueInstanceCount || totalUniquePartiallyMigratedCount === totalUniqueInstanceCount);
+  const internalCollectionOn = totalUniquePartiallyMigratedCount > 0;
+  const mightExist = get(setupModeData, 'detected.mightExist') || get(setupModeData, 'detected.doesExist');
+
+  let tooltip = null;
```

Comment:
Do we maybe need a default value for a `tooltip` in case none of the conditions below are true? 

I remember this was a problem somewhere else